lib/livebook/runtime/evaluator.ex

```diff
@@ -688,20 +692,100 @@ defmodule Livebook.Runtime.Evaluator do
     {result, code_markers}
   end
 
-  defp eval(:erlang, code, binding, env) do
+  # Erlang code is either statements as currently supported, or modules.
+  # In case we want to support modules - it makes sense to allow users to use
+  # includes, defines and thus we use the epp-module first - try to find out
+  #
+  # if in the tokens from erl_scan we find at least 1 module-token we assume
+  # that the user is defining a module, if not the previous code is called.
+
+  defp eval(:erlang, code, binding, env, tmp_dir) do
+    case :erl_scan.string(String.to_charlist(code), {1, 1}, [:text]) do
+      {:ok, tokens, _} ->
+        case :lists.keyfind(:module, 3, tokens) do
+          {:atom, _, _module} -> eval_erlang_module(code, binding, env, tmp_dir)
+          false -> eval_erlang_statements(code, tokens, binding, env)
+        end
+
+      {:error, {location, module, description}, _end_loc} ->
+        process_erlang_error(env, code, location, module, description)
+    end
+  end
+
+  # Explain to user: without tmp_dir to write files, they cannot compile erlang-modules
+  defp eval_erlang_module(_code, _binding, _env, nil) do
+    {{:error, :error, "erlang module compile needs tmp_dir", []}, []}
```

Comment:
```suggestion
    {{:error, :error, "writing Erlang modules require a writeable file system", []}, []}
```

Reply:
Let's also add a test for this error. It should be triggered if we don't put `@tag :with_ebin_path` on the given test :)

Reply:
Added test for this case.
I am getting some noise during mix-tests but none seems to be from the test-cases I added.